[Profiler] Clean up profiler includes.

robieta opened the pull request:
Summary: I've hit a lot of build issues in D32671972, and I've come to realize that a lot of it boils down to header hygene. `function.h` includes `profiler.h` *solely* to transitively include `record_function.h` which winds up leaking the profiler symbols. Moreover several files are relying on transitive includes to get access to `getTime`. As long as I have to touch all the places that use `getTime`, I may as well also move them to the new namespace.

Test Plan: Unit tests and CI.

Differential Revision: D32865907



cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @SciPioneer @H-Huang

albanD reviewed (approved):
SGTM

robieta commented:
@pytorchbot retest this please

aaronenyeshi commented on torch/csrc/jit/mobile/profiler_edge.cpp:
```diff
@@ -18,49 +18,52 @@ KinetoEdgeCPUProfiler::KinetoEdgeCPUProfiler(
+    auto post_processing =
+        [this, with_stack, with_modules](
+            std::vector<torch::autograd::profiler::KinetoEvent>& events) {
+          std::string no_debug_info(
```
nit: should indent be 2 spaces from line 31?

aaronenyeshi reviewed (approved):
LGTM, thank you for transferring some of the functions out of autograd namespace. Small nit, but won't block this on that.

Commit:
[Profiler] Clean up profiler includes. (#69421)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/69421

I've hit a lot of build issues in D32671972, and I've come to realize that a lot of it boils down to header hygene. `function.h` includes `profiler.h` *solely* to transitively include `record_function.h` which winds up leaking the profiler symbols. Moreover several files are relying on transitive includes to get access to `getTime`. As long as I have to touch all the places that use `getTime`, I may as well also move them to the new namespace.

Test Plan: Unit tests and CI.

Reviewed By: aaronenyeshi, albanD

Differential Revision: D32865907

fbshipit-source-id: 8dae6e23bcbe3ce8c13442fd80b06a72b5af180c
